### PR TITLE
Setting up CMake for kurobako

### PR DESCRIPTION
**Pull request by jonnerloller**
@added  Basic Structure of project and CMake file
@tofix  Linker for .lib file a bit wonky
@review Decide between auto adding files into projects vs manually adding files
[I support auto, only con is required to run CMake each time new files added]

### CODE REVIEW

File: src/kurobako/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+#Add manually each time.
+set (KUROBAKOSOURCEFILES 
+kurobakoconfig.h
+enginemain.h enginemain.cpp
+graphics/teststructure.cpp
+)
+
+#auto version
+#file (GLOB_RECURSE KUROBAKOSOURCEFILES *.h *.cpp)
```
Comment (Wagk) [inline src/kurobako/CMakeLists.txt]: Should automatically glob instead of hardcoding imo

File: src/kurobako/kurobakoconfig.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#ifndef KUROBAKOCONFIG_H
+#define KUROBAKOCONFIG_H
+
+#define KUROBAKO_VERSION_MAJOR
```
Comment (wagk) [inline src/kurobako/kurobakoconfig.h]: Is this supposed to be defined with a version number or something?

File: src/kurobako/CMakeLists.txt (hunk 1 above)
Comment (wagk) [inline src/kurobako/CMakeLists.txt]: Should glob automatically, Don't see us having automatic file generators anytime soon

File: src/sandbox/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+set(CMAKE_EXE_LINKER_FLAGS_PROFILE "/INCREMENTAL /MANIFEST /DEBUG /MACHINE:X64 /DYNAMICBASE kurobako_p.lib")
+set(CMAKE_EXE_LINKER_FLAGS_FINAL "/LTCG /MANIFEST /DEBUG /OPT:REF /MACHINE:X64 /DYNAMICBASE kurobako_final.lib")
+
+#Add manually each time.
+set (SANDBOXSOURCEFILES 
+main.cpp
+)
+
+#auto version
+#file (GLOB_RECURSE SANDBOXSOURCEFILES *.h *.cpp)
```
Comment (wagk) [inline src/sandbox/CMakeLists.txt]: Same question as above regarding globbing

**Comment by jonnerloller** [comment]
Btw if you are wondering why i am excluding x64, win32 folders, It's a current bug with CMake and VS2017. when the ZERO_BUILD runs, it drops artifacts all over the root folder

**Review by wagk — APPROVED** [review]

**Pull request merged**